### PR TITLE
wayfarer: init at 1.2.4

### LINKED ISSUE #248285: Package request: wayfarer
**Project description**

wayfarer is a screen recorder for GNOME/Wayland/Pipewire

**Metadata**

* homepage URL: https://github.com/stronnag/wayfarer
* source URL: https://github.com/stronnag/wayfarer
* license: gpl3.0
* platforms: linux

### PR DESCRIPTION
**Pull request by fgaz**
## Description of changes

Doesn't work for me on `sway` (similar failure as `kooha`), but maybe it works fine for GNOME users

Closes #248285

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/wa/wayfarer/package.nix [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    gst_all_1.gst-plugins-base
+    gst_all_1.gst-plugins-ugly
+    gtk4
+    libpulseaudio
+  ];
+
+  meta = with lib; {
+    description = "Screen recorder for GNOME / Wayland / pipewire";
+    homepage = "https://github.com/stronnag/wayfarer";
+    license = licenses.gpl3Only;
```
Comment (Aleksanaa) [inline pkgs/by-name/wa/wayfarer/package.nix:57]:
```suggestion
    license = licenses.gpl3Plus;
```

See https://github.com/stronnag/wayfarer?tab=readme-ov-file#miscellaneous

Comment (fgaz) [reply pkgs/by-name/wa/wayfarer/package.nix]: Good catch -- I shouldn't have trusted nix-init on this

File: pkgs/by-name/wa/wayfarer/package.nix [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    libpulseaudio
+  ];
+
+  meta = with lib; {
+    description = "Screen recorder for GNOME / Wayland / pipewire";
+    homepage = "https://github.com/stronnag/wayfarer";
+    license = licenses.gpl3Plus;
+    maintainers = with maintainers; [ fgaz ];
+    mainProgram = "wayfarer";
+    platforms = platforms.all;
```
Comment (Aleksanaa) [inline pkgs/by-name/wa/wayfarer/package.nix:60]: I'd suggest

```suggestion
    platforms = subtractLists platforms.darwin platforms.unix;
```

here, as this is targeting mutter and linux streaming stack (which can also possibly be found on other unixes, though). Building on darwin will not be useful at all.

**Review by Aleksanaa — APPROVED** [review]
Others lgtm, runs fine, can't record because I'm using hyprland :(

```
=> tree ./result/ -a
./result/
├── bin
│   ├── wayfarer
│   └── .wayfarer-wrapped
└── share
    ├── applications
    │   └── org.stronnag.wayfarer.desktop
    ├── glib-2.0
    ├── gsettings-schemas
    │   └── wayfarer-1.2.4
    │       └── glib-2.0
    │           └── schemas
    │               ├── gschemas.compiled
    │               └── org.stronnag.wayfarer.gschema.xml
    └── icons
        └── hicolor
            └── scalable
                └── apps
                    └── wayfarer.svg
```

**Comment by Aleksanaa** [comment]
Have you got anyone to test this? A friend of mine who uses gnome45 (unstable branch) said he couldn't use the screen recorder, but he didn't find any meaningful errors.

**Comment by Scrumplex** [comment]
Result of `nixpkgs-review pr 302772` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>wayfarer</li>
  </ul>
</details>

**Comment by Scrumplex** [comment]
Also doesn't work for me on Sway. It logs `pipewiresrc link fails`

**Comment by fgaz** [comment]
I added `pipewire` to `buildInputs` and that fixed it for me. I wonder why I didn't get any warning during `configurePhase`.

**Comment by Scrumplex** [comment]
Result of `nixpkgs-review pr 302772` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>wayfarer</li>
  </ul>
</details>

**Comment by Scrumplex** [comment]
Works on Sway!

**Pull request merged**